packages/transactions/src/sign.ts

```diff
@@ -1,11 +1,26 @@
 import { Signer } from '@near-js/signers';
 import sha256 from 'js-sha256';
 import BN from 'bn.js';
-import { serialize } from 'borsh';
 
-import { Action } from './actions';
+import { Action, SignedDelegate } from './actions';
 import { createTransaction } from './create_transaction';
-import { SCHEMA, Signature, SignedTransaction, Transaction } from './schema';
+import type { DelegateAction } from './delegated.js';
+import { encodeDelegateAction, encodeTransaction, SignedTransaction, Transaction } from './schema';
+import { Signature } from './signature';
+
+interface MessageSigner {
+    sign(message: Uint8Array): Promise<Uint8Array>;
```

Comment:
I'm not super attached to this, it was just my way of inching away from coupling transaction signing methods and network while avoiding stepping into the current state of keystores and their 1:1 account-network mapping. Definitely open to alternatives or tweaks to this approach.